docs/proposals/alternate-api-representations.md

```diff
@@ -0,0 +1,422 @@
+behavior is best represented as a variation on content-type. Supporting both is not limiting
+in the short term as long as we can validate correctly.
+
+
+### Example: Partial metadata retrieval
+
+The client may request to the server to return the list of namespaces as a
+`PartialObjectMetadata` kind, which is an object containing only `ObjectMeta` and
+can be serialized as protobuf or JSON. This is expected to be significantly more
+performant when controllers like the Garbage collector retrieve multiple objects.
+
+    GET /api/v1/namespaces
+    Accept: application/json;g=api.k8s.io,v=v1,as=PartialObjectMetadata, application/json
+
+The server would respond with
+
+    200 OK
+    Content-Type: application/json;g=api.k8s.io,v=v1,as=PartialObjectMetadata
+    {
+      "apiVersion": "api.k8s.io/v1",
+      "kind": "PartialObjectMetadataList",
+      "items": [
+        {
+          "apiVersion": "api.k8s.io/v1",
+          "kind": "PartialObjectMetadata",
+          "metadata": {
+            "name": "foo",
+            "resourceVersion": "10",
+            ...
+          }
+        },
+        ...
+      ]
+    }
+
+Note that the `as` parameter indicates to the server the Kind of the resource, but
+the Kubernetes API convention of returning a List with a known schema continues. An older
+server could ignore the presence of the `as` parameter on the media type and merely return
+a `NamespaceList` and the client would either use the content-type or the object Kind
+to distinguish. Because all responses are expected to be self-describing, an existing
+Kubernetes client would be expected to differentiate on Kind.
+
+An old server, not recognizing these parameters, would respond with:
+
+    200 OK
+    Content-Type: application/json
+    {
+      "apiVersion": "v1",
+      "kind": "NamespaceList",
+      "items": [
+        {
+          "apiVersion": "v1",
+          "kind": "Namespace",
+          "metadata": {
+            "name": "foo",
+            "resourceVersion": "10",
+            ...
+          }
+        },
+        ...
+      ]
+    }
+
+
+### Example: Retrieving a known version of the Scale resource
+
+Each API group that supports resources that can be scaled must expose a subresource on
+their object that accepts GET or PUT with a `Scale` kind resource. This subresource acts
+as a generic interface that a client that knows nothing about the underlying object can
+use to modify the scale value of that resource. However, clients *must* be able to understand
+the response the server provides, and over time the response may change and should therefore
+be versioned. Our current API provides no way for a client to discover whether a `Scale`
+response returned by `batch/v2alpha1` is the same as the `Scale` resource returned by
+`autoscaling/v1`.
+
+Under this proposal, to scale a generic resource a client would perform the following
+operations:
+
+    GET /api/v1/namespace/example/replicasets/test/scale
```

Comment:
nit: should be `/api/v1/namespace/example/replicationcontrollers` or `/apis/extensions/v1beta1/namespace/example/replicasets`